app/validation/validator.py

```diff
@@ -122,80 +122,77 @@ def validate_answer(self, block, answer):
 
     def validate_range_types_from_answers(self, json_to_validate):
         errors = []
+        answer_ranges = {}
```

Comment:
Is `answer_ranges` necessary? The `answer_range` returned by `self._get_answer_ranges()` never gets used outside the scope of the loop. So maybe remove this and replace `answer_ranges[answer.get('id')]` with a single `answer_range` variable.

Reply:
I missed that fact that `_get_answer_range_values()` has `answer_ranges` passed through to it. And that seems to be to follow the `answer_id` and get the min/max values of a related answer (in `_get_answer_minimum_or_maximum()`).

Since `answer_ranges` is only getting built up within this loop there's an assumption that if `_get_answer_range_values` needs to reference another answer then that answer will already have been looped over. I.e. it assumes that answers are in order and that only previous answers will ever be referenced.

This sounds like it's taken into account but I thought I'd ask the question. @jonnyshaw89 might know?

Reply:
this is the case @nealedj, hence why we pass the ranges through